### PR TITLE
Bump cOS-toolkit to latest version 

### PR DESCRIPTION
**Pull request by bk201**
Bump packages in cOS-toolkit to the latest version by switching repo.

- Switch to quay.io/costoolkit/releases-green repo.
  - rancherd is updated to v0.0.1-alpha07, which contains RKE2 fix.
   (workaround https://github.com/harvester/harvester-installer/blob/a69dc07f43fe54df219078b8a30564dd97289bae/package/harvester-os/Dockerfile#L7
    is no longer needed)
- Use new disk layout mechanism to create persistent bind mounts.
- Move more customization from harvester-installer to here.

### CODE REVIEW

**Review by guangbochen — APPROVED** [review]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -75,6 +75,18 @@ RUN zypper in -y \
     vim \
     which
 
+# Harvester needs these packages
+RUN zypper in -y apparmor-parser \
```
Comment (gitlawr) [inline Dockerfile:79]: Why don't we put this in the Dockerfile for harvester-os?

Comment (bk201) [reply Dockerfile:79]: Both works, putting it here saves some build time in harvester-installer.

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**